### PR TITLE
KAFKA-16921: migrate connect module to junit 5

### PR DESCRIPTION
**Pull request by FrankYang0529**
* org.apache.kafka.connect.storage
* org.apache.kafka.connect.test.util
* org.apache.kafka.connect.util

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/SharedTopicAdminTest.java [hunk 1]
```diff
@@ -46,14 +48,10 @@ public class SharedTopicAdminTest {
     @Mock private Function<Map<String, Object>, TopicAdmin> factory;
     private SharedTopicAdmin sharedAdmin;
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/util/SharedTopicAdminTest.java]: we can move this to local variable, right?

File: connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java [hunk 2]
```diff
@@ -123,12 +127,23 @@ public void testSaveRestore() throws Exception {
...
+        store = new FileOffsetBackingStore(converter);
+        tempFile = assertDoesNotThrow(() -> File.createTempFile("fileoffsetbackingstore", null));
+        Map<String, String> props = new HashMap<>();
+        props.put(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG, tempFile.getAbsolutePath());
+        props.put(StandaloneConfig.KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+        props.put(StandaloneConfig.VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+        config = new StandaloneConfig(props);
+        store.configure(config);
+        store.start();
         assertTrue(((ThreadPoolExecutor) store.executor).getThreadFactory()
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/storage/FileOffsetBackingStoreTest.java]: Maybe we can move this `assert` to `setup` and then remove this test case. That is more simple to me

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thanks for this patch. two comments are left.

**Comment by FrankYang0529** [comment]
Hi @chia7712, thanks for your review. I addressed all comments. May you take a look? Thank you.

**Pull request merged**